impl/authorization/authorization.go

```diff
@@ -65,7 +63,16 @@ func resourceLabel(mapID, appID int64) string {
 	return fmt.Sprintf("%d|%d", mapID, appID)
 }
 
-func permitted(role *authzpb.AuthorizationPolicy_Role, permission authzpb.Permission) bool {
+func isPrincipalInRole(role *authzpb.AuthorizationPolicy_Role, identity string, permission authzpb.Permission) bool {
+	for _, p := range role.GetPrincipals() {
+		if p == identity && isPermisionInRole(role, permission) {
```

Comment:
Consider removing isPermissionInRole from this function, and having the caller test both 
`isPrincipalInRole() && isPermissionInRole()`